packages/stencil-library/src/components/button/button.tsx

```diff
@@ -1,22 +1,18 @@
 import { Component, Prop, h } from "@stencil/core";
-import { Host, HTMLStencilElement } from "@stencil/core/internal";
-import { Element } from '@stencil/core';
-import { Mode } from "../../interface";
-
-
-export type ButtonVariant = "primary" | "secondary" | "link" | "danger"
+import { ButtonVariant, Mode, IconPosition } from "../../interface";
 
 @Component({
     tag: "njwds-button",
 })
 export class Button {
     @Prop() variant: ButtonVariant = "primary";
```

Comment:
[sand] I found out the other day there's a way to mark [required properties in Stencil](https://stenciljs.com/docs/properties#required-properties) for TSX, might be worth adding to the `variant` and `mode` properties!

Reply:
Nice, good find. In this case I think we can skip using it because doing so would result in the ts error `Declarations with initializers cannot also have definite assignment assertions`. I imagine we want to allow the user to not include it and just initialize a default value here, but I definitely see this being a good thing to do in other instances.